CRM/Utils/System.php

```diff
@@ -306,8 +306,14 @@ public static function url(
    * @return string|null
    */
   public static function getUrlPath() {
-    if (isset($_GET[CRM_Core_Config::singleton()->userFrameworkURLVar])) {
-      return $_GET[CRM_Core_Config::singleton()->userFrameworkURLVar];
+    $config = CRM_Core_Config::singleton();
+
+    if (method_exists($config->userSystem, 'getUrlPath')) {
+      return $config->userSystem->getUrlPath();
+    }
+
+    if (isset($_GET[$config->userFrameworkURLVar])) {
+      return $_GET[$config->userFrameworkURLVar];
     }
     return NULL;
   }
```

Comment:
I have a small `r-code` suggestion to make this polymorphism more conventional. Will push up onto this PR in a few minutes after `r-running` on D7 and writing commit notes.

Reply:
Pushed up. Tested on D7 to ensure that the refactoring was generally well-formed. Discussed more in commit notes.

Reply:
Looks good. Thank you!